metricbeat/module/jolokia/jmx/config.go

```diff
@@ -103,41 +110,326 @@ func (m AttributeMapping) Get(mbean, attr string) (Attribute, bool) {
 	return a, found
 }
 
-// Parse strings with properties with the format key=value, being:
-// - key a nonempty string of characters which may not contain any of the characters,
-//   comma (,), equals (=), colon, asterisk, or question mark.
-// - value a string that can be quoted or unquoted, if unquoted it cannot be empty and
-//   cannot contain any of the characters comma, equals, colon, or quote.
-var propertyRegexp = regexp.MustCompile("[^,=:*?]+=([^,=:\"]+|\".*\")")
+// MBeanName is an internal struct used to store
+// the information by the parsed `mbean` (bean name) configuration
+// field in `jmx.mappings`.
+type MBeanName struct {
+	Domain     string
+	Properties map[string]string
+}
 
-func canonicalizeMBeanName(name string) (string, error) {
-	// From https://docs.oracle.com/javase/8/docs/api/javax/management/ObjectName.html#getCanonicalName--
-	//
-	//   Returns the canonical form of the name; that is, a string representation where the
-	//   properties are sorted in lexical order.
-	//   The canonical form of the name is a String consisting of the domain part,
-	//   a colon (:), the canonical key property list, and a pattern indication.
-	//
-	parts := strings.SplitN(name, ":", 2)
+var mbeanRegexp = regexp.MustCompile("([^,=:*?]+)=([^,=:\"]+|\".*\")")
+
+// This replacer is responsible for adding a "!" before special characters in GET request URIs
+// For more information refer: https://jolokia.org/reference/html/protocol.html
+var mbeanGetEscapeReplacer = strings.NewReplacer("\"", "!\"", ".", "!.", "!", "!!", "/", "!/")
```

Comment:
Btw, dot doesn't appear in the [escaping rules table](https://jolokia.org/reference/html/protocol.html#escape-rules), I don't think it does any harm to escape it in any case.

Reply:
Hi @jsoriano ! Indeed it does not appear in  [escaping rules table](https://jolokia.org/reference/html/protocol.html#escape-rules), but I had to escape it because as far as I remember it threw errors otherwise.